var/spack/repos/builtin/packages/legion/package.py

```diff
@@ -92,7 +92,7 @@ class Legion(CMakePackage, ROCmPackage):
     patch("hip-offload-arch.patch", when="@23.03.0 +rocm")
 
     def patch(self):
-        if "network=gasnet conduit=ofi-slingshot11 ^cray-mpich+wrappers" in self.spec:
+        if self.spec.satisfies("network=gasnet conduit=ofi-slingshot11 ^cray-mpich+wrappers"):
```

Comment:
```suggestion
        if self.spec.satisfies("network=gasnet conduit=ofi-slingshot11 ^[virtuals=mpi] cray-mpich+wrappers"):
```
and fix style likely.